### PR TITLE
Update dependencies to newest versions

### PR DESCRIPTION
**Pull request by roypat**
## Changes

Updates all our dependencies to their newest versions. Followup PR to #3248 (which should be merged first)

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license.

## PR Checklist

- [x] All commits in this PR are signed (`git commit -s`).
- [x] If a specific issue led to this PR, this PR closes the issue.
- [x] The description of changes is clear and encompassing.
- [x] Any required documentation changes (code and docs) are included in this PR.
- [x] New `unsafe` code is documented.
- [x] API changes follow the [Runbook for Firecracker API changes][2].
- [x] User-facing changes are mentioned in `CHANGELOG.md`.
- [x] All added/changed functionality is tested.

---

- [ ] This functionality can be added in [`rust-vmm`][1].

[1]: https://github.com/rust-vmm
[2]: https://github.com/firecracker-microvm/firecracker/blob/main/docs/api-change-runbook.md

### CODE REVIEW

**Review by dianpopa — COMMENTED** [review]
LGTM, will approve once #3248 is merged.

**Review by luminitavoicu — DISMISSED** [review]

**Review by dianpopa — DISMISSED** [review]

**Review by luminitavoicu — APPROVED** [review]

**Review by dianpopa — APPROVED** [review]

**Pull request merged**
